### PR TITLE
Traxis log-parsing job with QueryMetrics

### PR DESCRIPTION
**Pull request by EgorAfutin**

### CODE REVIEW

File: src/common/log_parsing/metadata.py [hunk 1]
```diff
@@ -58,7 +58,7 @@ class AbstractField:
     """
     __metaclass__ = ABCMeta
 
-    def __init__(self, name, output_name=None):
+    def __init__(self, name="any_field", output_name=None):
```
Comment (halfwhite) [inline src/common/log_parsing/metadata.py]: What is "any_field"?

File: src/common/log_parsing/dict_event_creator/key_value_parser.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        :param text: input text
+        :return: parsed dict
+        :raises: ParsingException when text doesn't confirm key/value format
+        """
+
+        def __trim_and_transform_to_key_value(key_value):
+            pair = key_value.split("=")
+            if len(pair) != 2 | pair[0].isspace():
+                raise ParsingException("Can't parse message!")
+            return re.sub('(?<!^)(?=[A-Z])', '_', pair[0].strip()).lower(), pair[1].strip()
```
Comment (halfwhite) [inline src/common/log_parsing/dict_event_creator/key_value_parser.py]: I believe the parser shall not transform a key.

File: src/common/log_parsing/dict_event_creator/key_value_parser.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+        return dict(filter(lambda value: value[1], map(__trim_and_transform_to_key_value, text.split(","))))
```
Comment (halfwhite) [inline src/common/log_parsing/dict_event_creator/key_value_parser.py]: Separator character should be configurable. e.g. via constructor.

File: src/common/log_parsing/dict_event_creator/key_value_parser.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    @staticmethod
+    def parse(text):
+        """
+        Parses text using pattern
+        :param text: input text
+        :return: parsed dict
+        :raises: ParsingException when text doesn't confirm key/value format
+        """
+
+        def __trim_and_transform_to_key_value(key_value):
```
Comment (halfwhite) [inline src/common/log_parsing/dict_event_creator/key_value_parser.py]: why a function inside function is better than just 2 separate functions?

File: src/applications/log_parsing/traxis_frontend/driver.py [hunk 5]
```diff
@@ -74,10 +76,56 @@ def create_event_creators(configuration):
             return_empty_dict=True),
         matcher=SubstringMatcher("Cannot purchase products of type"))
 
+    query_metrics_event_creator_with_request_id_and_customer_id = EventCreator(
+        Metadata(
+            [StringField("query_metrics"),
+             StringField("request_id"),
+             StringField("customer_id")]),
+        RegexpParser(
+            r"^.*\[RequestId\s*=\s*(?P<request_id>.*)\]\s*\[CustomerId\s*=\s*(?P<customer_id>.*)\]\s*"
```
Comment (needmorec) [inline src/applications/log_parsing/traxis_frontend/driver.py]: please remove duplicated event creators and use only one with optional groups

File: src/applications/log_parsing/traxis_frontend/driver.py [hunk 6]
```diff
@@ -74,10 +76,56 @@ def create_event_creators(configuration):
             return_empty_dict=True),
         matcher=SubstringMatcher("Cannot purchase products of type"))
 
+    query_metrics_event_creator_with_request_id_and_customer_id = EventCreator(
```
Comment (halfwhite) [inline src/applications/log_parsing/traxis_frontend/driver.py]: It becomes to big method. Pls, extract creating event creators in separate methods.

File: src/test/unit/common/log_parsing/dict_event_creator/test_key_value_matches_pattern.py [hunk 7]
```diff
@@ -0,0 +1,41 @@
+import unittest
+
+from common.log_parsing.metadata import ParsingException
+from common.log_parsing.dict_event_creator.key_value_parser import KeyValueParser
+
+
+class KeyValueParserTestCase(unittest.TestCase):
```
Comment (halfwhite) [inline src/test/unit/common/log_parsing/dict_event_creator/test_key_value_matches_pattern.py]: Add check for an input without spaces and partially with spaces.

File: src/applications/log_parsing/traxis_frontend/key_value_event_creator.py [hunk 8]
```diff
@@ -0,0 +1,23 @@
+"""
+Creator for traxis_frontend:
+- KeyValueEventCreator
+"""
+from common.log_parsing.dict_event_creator.event_creator import EventCreator
+
+
+class KeyValueEventCreator(EventCreator):
```
Comment (halfwhite) [inline src/applications/log_parsing/traxis_frontend/key_value_event_creator.py]: This code does nothing specific to key value processing. Consider renaming the class.

File: src/applications/log_parsing/traxis_frontend/key_value_event_creator.py [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+       Converts row to typed values according metadata.
+       :param row: input row
+       :param context: dictionary with additional data.
+       :return: map representing event where key is event field name and value is field value.
+       :exception ParsingException if converting goes wrong.
+       """
+        parsed = self._parser.parse(row[self.get_field_to_parse()])
+        return {
+            field:
+                self._metadata.get_field_by_name("any_field").get_value(value, context)
```
Comment (halfwhite) [inline src/applications/log_parsing/traxis_frontend/key_value_event_creator.py]: "any_field" is a magic string.

**Pull request merged**